infra/modules/aws-psoxy-bulk/main.tf

```diff
@@ -98,31 +81,55 @@ resource "aws_s3_bucket" "output" {
   bucket = "psoxy-${var.instance_id}-output"
 }
 
-module "psoxy-file-handler" {
-  source = "../aws-psoxy-instance"
+module "psoxy_lambda" {
+  source = "../aws-psoxy-lambda"
 
   function_name        = "psoxy-${var.instance_id}"
   handler_class        = "co.worklytics.psoxy.S3Handler"
-  source_kind          = var.source_kind
-  path_to_function_zip = module.psoxy-package.path_to_deployment_jar
-  function_zip_hash    = module.psoxy-package.deployment_package_hash
-  path_to_config       = "${var.psoxy_base_dir}/configs/${var.source_kind}.yaml"
+  timeout_seconds      = 600 # 10 minutes
+  memory_size_mb       = 512
+  path_to_config       = var.path_to_config
+  path_to_function_zip = var.path_to_function_zip
+  function_zip_hash    = var.function_zip_hash
   aws_assume_role_arn  = var.aws_assume_role_arn
-  example_api_calls    = [] #None, as this function is called through the S3 event
-  aws_account_id       = var.aws_account_id
+  source_kind          = var.source_kind
+  parameters           = []
+}
 
-  parameters = []
 
-  environment_variables = {
-    INPUT_BUCKET  = aws_s3_bucket.input.bucket,
-    OUTPUT_BUCKET = aws_s3_bucket.output.bucket
+
+resource "aws_lambda_function" "psoxy-instance" {
```

Comment:
Duplicated with resource it will be created on L85. Adding something like the following after L96 I think it can work:

```
environment_variables = {
        INPUT_BUCKET  = aws_s3_bucket.input.bucket,
        OUTPUT_BUCKET = aws_s3_bucket.output.bucket
      }
```